week2/internal/repository/implement/film_repository.go

```diff
@@ -89,3 +91,30 @@ func (repo *FilmRepository) DeleteFilm(ctx context.Context, id int64) error {
 		return nil
 	})
 }
+
+func(repo *FilmRepository) CreateFilm(ctx context.Context, film *entity.Film) error {
+	err := repo.db.WithContext(ctx).Create(film).Error
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func(repo *FilmRepository) UpdateFilm(ctx context.Context, film *entity.Film) error {
+	var exsitingFilm entity.Film
+
+	// find film by ID first
+	err := repo.db.First(&exsitingFilm, film.ID).Error
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound){
+			return fmt.Errorf("film with ID %d not found", film.ID)
+		}
+		return err
+	}
+	// update film
+	err = repo.db.WithContext(ctx).Save(film).Error
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
What if after I select, another thread delete the resource. Can you confirm that the Save operation doesn't create a new record ?